File: src/main/java/de/steklopod/sec03/Lec01FluxCreate.java
package de.steklopod.sec03;

import de.steklopod.courseutil.Util;
import reactor.core.publisher.Flux;

public class Lec01FluxCreate {

    public static void main(String[] args) {

        Flux.create(fluxSink -> {
            String country;
            do{
                country = Util.faker().country().name();
                fluxSink.next(country);
            }while (!country.toLowerCase().equals("canada"));
            fluxSink.complete();
        })
        .subscribe(Util.subscriber());



    }

}

File: src/main/kotlin/de/steklopod/sec03/Lec01FluxCreate.kt
package de.steklopod.sec03

import de.steklopod.utils.Util.faker
import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux
import reactor.core.publisher.FluxSink
import java.util.*

object Lec01FluxCreate {
    @JvmStatic
    fun main(args: Array<String>) {
        Flux.create { fluxSink: FluxSink<String> ->
            var country: String
            do {
                country = faker().country().name()
                fluxSink.next(country)
            } while (country.lowercase(Locale.getDefault()) != "canada")
            fluxSink.complete()
        }
            .subscribe(subscriber())
    }
}
